### PR TITLE
Add sender level tags

### PR DESCRIPTION
**Pull request by turtlespin**
# Overview

This PR adds the concept of tags which can be set once against a RaygunSender instance and then applied to each exception which is raised through that sender. This avoids the requirement to specify tags with each invocation of **send_exception**. 
A new method **set_tags** has been added to RaygunSender to allow these tags to be specified.

I have also updated the API endpoint to use api.raygun.com rather than api.raygun.io as part of this PR.

# Examples

## Setup

`client = raygunprovider.RaygunSender('YOUR_API_KEY')`

Initialize a client instance

`client.set_tags(["process_tag1", "process_tag2"])`

Use the new set_tags method to apply tags at a sender level

## Example 1

`client.send_exception(exc_info=(exc_type, exc_value, exc_traceback))`

Raising an exception with no tags specified

The resulting error has the sender level tags applied

## Example 2

`client.send_exception(exc_info=(exc_type, exc_value, exc_traceback), tags={"tag1", "tag2"})`

Raising an exception with tags specified

The resulting error has both the sender level tags and the instance level tags applied

## Example 3

`client.send_exception(exc_info=(exc_type, exc_value, exc_traceback), tags={"tag1", "tag1"})`
`client.send_exception(exc_info=(exc_type, exc_value, exc_traceback), tags={"tag3", "tag4"})`

Raising 2 exceptions with different instance level tags specified

The 2 resulting errors both have the sender level tags applied but differ with the instance level tags that are applied

### CODE REVIEW

**Comment by darcythomas** [comment]
Side note: Do we have a matrix of features (like this) vs providers.

Something we could use to make small improvements to the providers, during our downtime etc.

File: python2/raygun4py/raygunmsgs.py [hunk 1]
```diff
@@ -90,8 +90,10 @@ def set_customdata(self, user_custom_data):
         return self
 
     def set_tags(self, tags):
-        if type(tags) is list or tuple:
-            self.raygunMessage.details['tags'] = tags
+        if type(tags) is list:
```
Comment (darcythomas) [inline python2/raygun4py/raygunmsgs.py:93]: We are not checking for tuples here anymore.
Is that a breaking change, or can tuples be coerced in to a list in python?

File: python2/raygun4py/raygunprovider.py [hunk 2]
```diff
@@ -31,8 +31,9 @@ class RaygunSender:
 
     api_key = None
     endpointprotocol = 'https://'
-    endpointhost = 'api.raygun.io'
+    endpointhost = 'api.raygun.com'
     endpointpath = '/entries'
+    process_tags = None
```
Comment (darcythomas) [inline python2/raygun4py/raygunprovider.py:36]: I am not a fan of nulls (`None`) could this be an empty array (list?) instead.

File: python2/raygun4py/raygunprovider.py [hunk 3]
```diff
@@ -31,8 +31,9 @@ class RaygunSender:
 
     api_key = None
     endpointprotocol = 'https://'
-    endpointhost = 'api.raygun.io'
```
Comment (darcythomas) [inline python2/raygun4py/raygunprovider.py:34]: Do we also  need to include release notes so people can change firewall rules?

Comment (turtlespin) [reply python2/raygun4py/raygunprovider.py:34]: Good thought, I might pull this change out as a separate PR and we can discuss this as a more general update across providers.

Comment (turtlespin) [reply python2/raygun4py/raygunprovider.py:34]: Have reverted this change, will address in a later PR

Comment (turtlespin) [reply python2/raygun4py/raygunprovider.py:34]: Have reverted this change, will address in a later PR

**Review by darcythomas — APPROVED** [review]

**Pull request merged**